src/services/PlateGenerationService.ts

```diff
@@ -0,0 +1,209 @@
+import {AWSError, config as AWSConfig, Lambda} from "aws-sdk";
+import {Service} from "../models/injector/ServiceDecorator";
+import {LambdaService} from "./LambdaService";
+import moment from "moment";
+import {
+    ICertificatePayload,
+    IGeneratedCertificateResponse,
+    IInvokeConfig,
+    IMOTConfig,
+    ITechRecord,
+    ITechRecordWrapper
+} from "../models";
+import {PromiseResult} from "aws-sdk/lib/request";
+import {VEHICLE_TYPES} from "../models/Enums";
+
+/**
+ * Service class for Plate Generation
+ */
+@Service()
+class PlateGenerationService {
+    private readonly s3Client: S3BucketService;
+    private readonly config: Configuration;
+    private readonly lambdaClient: LambdaService;
+
+    constructor(s3Client: S3BucketService, lambdaClient: LambdaService) {
+        this.s3Client = s3Client;
+        this.config = Configuration.getInstance();
+        this.lambdaClient = lambdaClient;
+
+        AWSConfig.lambda = this.config.getInvokeConfig().params;
+    }
+
+    /**
+     * Generates Plates certificate for a given tech record
+     * @param vehicle - source tech record for plate generation
+     */
+    public async generateCertificate(vehicle: ITechRecordWrapper): Promise<IGeneratedCertificateResponse> {
+        const config: IMOTConfig = this.config.getMOTConfig();
+        const iConfig: IInvokeConfig = this.config.getInvokeConfig();
+        const techRecord: ITechRecord = vehicle.techRecord[0];
+        const platePayload: string = JSON.stringify(await this.generatePayload(vehicle));
+        console.log("STRING PAYLOAD", platePayload);
+
+        const invokeParams: any = {
+            FunctionName: iConfig.functions.plateGen.name,
+            InvocationType: "RequestResponse",
+            LogType: "Tail",
+            Payload: JSON.stringify({
+                httpMethod: "POST",
+                pathParameters: {
+                    documentName: config.documentNames.vtg6_vtg7,
+                    documentDirectory: config.documentDir
+                },
+                json: true,
+                body: platePayload
+            }),
+        };
+
+        return this.lambdaClient.invoke(invokeParams)
+            .then((response: PromiseResult<Lambda.Types.InvocationResponse, AWSError>) => {
+                const payload: any = this.lambdaClient.validateInvocationResponse(response);
+                const resBody: string = payload.body;
+                const responseBuffer: Buffer = Buffer.from(resBody, "base64");
+                return {
+                    vin: vehicle.vin,
+                    vrm: vehicle.primaryVrm ? vehicle.primaryVrm : vehicle.trailerId,
+                    dateOfIssue: moment(techRecord.plates.plateIssueDate).format("DD MMMM YYYY"),
+                    certificateType: config.documentNames.vtg6_vtg7.split(".")[0],
+                    fileFormat: "pdf",
+                    fileName: `${vehicle.vin}_${techRecord.plates.plateSerialNumber}.pdf`,
```

Comment:
Is this naming convention mentioned in the ticket  as I was not able to see it.

Reply:
Yes, AC6: AND the S3 key of this PDF file is a vin_plateSerialNumber